Many canary tests are intermittently failing while waiting or confirming ceph is running

<!-- **Are you in the right place?**
1. For issues or feature requests, please create an issue in this repository.
2. For general technical and non-technical questions, we are happy to help you on our [Rook.io Slack](https://slack.rook.io/).
3. Did you already search the existing open issues for anything similar? -->

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Canary integration tests are failing intermittently, causing instability in the CI.

In the most recent canary CI runs in the master branch in the past week, here is the number of occurrences for each test failure. Issues for the most common three failures will be opened separately. Let's track the lesser often failures with this issue.

2- canary
3- raw-disk
5- two-osds-in-device
4- osd-with-metadata-device
4- encryption
1- encryption-pvc-db
4- lvm
1- pvc-db
1- pvc-db-wal
11- encryption-pvc-kms-vault-token
20- multi-cluster-mirroring
10- rgw-multisite-testing
2- multus-cluster-network
3- csi-hostnetwork-disabled

Some examples of failures:

While waiting for ceph to start in [raw disk](https://github.com/rook/rook/actions/runs/4186190468/jobs/7254342557):
```
+ echo 'Ceph is up and running, have a look!'
+ kubectl -n rook-ceph exec rook-ceph-tools-7c4b8bb9b5-t[86](https://github.com/rook/rook/actions/runs/4186190468/jobs/7254344366#step:8:87)mk -- ceph --connect-timeout 3 -s
E0215 17:03:33.2024[87](https://github.com/rook/rook/actions/runs/4186190468/jobs/7254344366#step:8:88)   64910 memcache.go:255] couldn't get resource list for external.metrics.k8s.io/v1beta1: Got empty response for: external.metrics.k8s.io/v1beta1
timed out
command terminated with exit code 1
Error: Process completed with exit code 1.
```

Waiting for ceph to start in [this test](https://github.com/rook/rook/actions/runs/4180309083/jobs/7241164384):
```
+ return current status: cluster: id: cbd7b99c-1b12-4415-8d41-61341702565c health: HEALTH_WARN Reduced data availability: 20 pgs inactive services: mon: 1 daemons, quorum a '(age' '7m)' mgr: 'a(active,' since '6m)' mds: 1/1 daemons up, 1 hot standby osd: 2 osds: 1 up '(since' '5m),' 2 in '(since' '6m)' cephfs-mirror: 1 daemon active '(1' 'hosts)' rbd-mirror: 1 daemon active '(1' 'hosts)' data: volumes: 1/1 healthy pools: 13 pools, 170 pgs objects: 32 objects, 14 KiB usage: 22 MiB used, 14 GiB / 14 GiB avail pgs: 11.765% pgs unknown 150 active+clean 20 unknown io: client: 639 B/s rd, 1 op/s rd, 0 op/s wr progress: Global Recovery Event '(5m)' '[========================....]' '(remaining:' '42s)'
tests/scripts/validate_cluster.sh: line 58: return: current: numeric argument required
```

**Expected behavior:**
Passing tests

--- a/.github/workflows/canary-integration-test.yml
+++ b/.github/workflows/canary-integration-test.yml
@@ -882,6 +882,9 @@ jobs:
           timeout 120 bash -c 'while kubectl -n rook-ceph get cephobjectstore my-store; do echo "waiting for objectstore my-store to delete"; sleep 5; done'
           kubectl create -f tests/manifests/test-object.yaml
           tests/scripts/validate_cluster.sh rgw
+          # the test is already waiting for the rgw daemon to start, but this is just a hack until we find
+          # what else we need to wait for
+          sleep 30
           tests/scripts/deploy-validate-vault.sh validate_rgw
 
       - name: collect common logs

--- a/.mergify.yml
+++ b/.mergify.yml
@@ -207,7 +207,6 @@ pull_request_rules:
       - "check-success=encryption-pvc-kms-vault-token-auth"
       - "check-success=encryption-pvc-kms-vault-k8s-auth"
       - "check-success=lvm-pvc"
-      - "check-success=multi-cluster-mirroring"
       - "check-success=rgw-multisite-testing"
       - "check-success=TestCephSmokeSuite (v1.21.14)"
       - "check-success=TestCephSmokeSuite (v1.26.1)"

--- a/tests/scripts/collect-logs.sh
+++ b/tests/scripts/collect-logs.sh
@@ -10,7 +10,7 @@ set -x
 LOG_DIR="${LOG_DIR%/}" # remove trailing slash if necessary
 mkdir -p "${LOG_DIR}"
 
-CEPH_CMD="kubectl -n ${CLUSTER_NAMESPACE} exec deploy/rook-ceph-tools -- ceph --connect-timeout 3"
+CEPH_CMD="kubectl -n ${CLUSTER_NAMESPACE} exec deploy/rook-ceph-tools -- ceph --connect-timeout 10"
 
 $CEPH_CMD -s >"${LOG_DIR}"/ceph-status.txt
 $CEPH_CMD osd dump >"${LOG_DIR}"/ceph-osd-dump.txt

--- a/tests/scripts/validate_cluster.sh
+++ b/tests/scripts/validate_cluster.sh
@@ -25,7 +25,7 @@ OSD_COUNT=$2
 #############
 # FUNCTIONS #
 #############
-EXEC_COMMAND="kubectl -n rook-ceph exec $(kubectl get pod -l app=rook-ceph-tools -n rook-ceph -o jsonpath='{.items[*].metadata.name}') -- ceph --connect-timeout 3"
+EXEC_COMMAND="kubectl -n rook-ceph exec $(kubectl get pod -l app=rook-ceph-tools -n rook-ceph -o jsonpath='{.items[*].metadata.name}') -- ceph --connect-timeout 10"
 
 function wait_for_daemon () {
   timeout=90

--- a/tests/scripts/validate_cluster.sh
+++ b/tests/scripts/validate_cluster.sh
@@ -55,7 +55,10 @@ function test_demo_mgr {
 
 function test_demo_osd {
   # shellcheck disable=SC2046
-  return $(wait_for_daemon "$EXEC_COMMAND -s | grep -sq \"$OSD_COUNT osds: $OSD_COUNT up.*, $OSD_COUNT in.*\"")
+  ret_val=$(wait_for_daemon "$EXEC_COMMAND -s | grep -sq \"$OSD_COUNT osds: $OSD_COUNT up.*, $OSD_COUNT in.*\"")
+  # debug info for an intermittent failure
+  echo "Return value = $ret_val"
+  return $ret_val
 }
 
 function test_demo_rgw {

